pkg/engine/plan/plan.go

```diff
@@ -933,6 +1008,25 @@ func (v *Visitor) resolveFieldPath(ref int) []string {
 	return []string{fieldName}
 }
 
+func (v *Visitor) resolveTransformation(ref int) *pipe.Pipeline {
```

Comment:
I put it as an independent function, but for performance could be done as part of the `resolveFieldPath` function